lib/net.js

```diff
@@ -906,8 +906,11 @@ Socket.prototype.connect = function(options, cb) {
     if (localPort && !util.isNumber(localPort))
       throw new TypeError('localPort should be a number: ' + localPort);
 
-    if (port <= 0 || port > 65535)
-      throw new RangeError('port should be > 0 and < 65536: ' + port);
+    if (options.port !== undefined && parseInt(options.port, 10) !== port)
+      throw new TypeError('port should be a number: ' + options.port);
```

Comment:
We might also want to change the error message to `'port should be undefined or should represent a valid number: ' + options.port'`, but I'm not sure how much of a breaking change that would be for users who consume error messages.